blenderproc/python/loader/Front3DLoader.py

```diff
@@ -416,4 +419,32 @@ def _move_and_duplicate_furniture(data: dict, all_loaded_furniture: list) -> Lis
                             rotation_mat = mathutils.Quaternion(child["rot"]).to_euler().to_matrix().to_4x4()
                             # transform it into the blender coordinate system and then to an euler
                             new_obj.set_rotation_euler((blender_rot_mat @ rotation_mat).to_euler())
+                            
+                            if new_obj.get_cp("uid") == prev_uid:
+                                equal_objs[-1].append(new_obj)
+                            else:
+                                equal_objs.append([new_obj])
+                                prev_uid = new_obj.get_cp("uid")
```

Comment:
There is no guarantee that the objects which are next to each other in the 3D Front file are from the same object. This could also mean that two chairs which come right after each other are merged to one.

Reply:
sadly, doing the merging while loading is too big of a mess for me now bc it breaks down this method due to missing references/objects. btw, this seems to work correctly

Reply:
Anyway, from what I could've seen this works correctly because of the way the original array is being built.

Reply:
An idea would be to change `extend` to append. In that way, you have the same vector but not flattened. Thus you retain all references and pieces of the objects and can merge it afterwards.